typescript/common-resolvers/src/image.ts

```diff
@@ -220,6 +221,29 @@ const createImage = async (
   return createdImage;
 };
 
+const deleteImage = async (
+  _: any,
+  args: MutationDeleteImageArgs,
+  { repository, user }: Context
+): Promise<DbImage> => {
+  const imageId = args.where.id;
+  const imageToDelete = await throwIfResolvesToNil(
+    "No image with such id",
+    repository.image.get
+  )({ id: imageId }, user);
+  const labelsToDelete = await repository.label.list({
+    imageId,
+    user,
+  });
+  await Promise.all(
+    labelsToDelete.map((label) =>
+      repository.label.delete({ id: label.id }, user)
+    )
+  );
+  await repository.image.delete({ id: imageId }, user);
+  return imageToDelete;
+};
+
```

Comment:
Shouldn't you delete the image in the cache as well, calling `await repository.upload.delete(image.url)`?

Reply:
I agree I think

Reply:
Done in 9ad999fbcf54906b60f0c3ae4fd0fa1fb414fb3c

As a side note I added a check to detect if the image that is being deleted is not one from the tutorial dataset